### ISSUE TITLE
Undo feature is not working for 'Read only' notebooks

### ISSUE THREAD

**Issue by Prateek032** [att/rcloud#1899]
Open other user Notebook and press < Ctrl + Z > (undo), previous History version of the notebook is not opening

**Comment by gordonwoodhull** [comment]
Confirmed.

**Comment by gordonwoodhull** [comment]
It is kind of weird to undo a readonly notebook, but since we are already abusing a well-known feature, we should be consistent about it.

**Comment by Prateek032** [comment]
This issue still persist on latest bae8152 commit

**Comment by gordonwoodhull** [comment]
So #1976 / #1971 interferes here. Looks like undo and redo commands should be available in edit mode but not view mode. It's not about the notebook status, just the page that is open, for this one.

**Comment by gordonwoodhull** [comment]
Thanks @Prateek032!

**Comment by gordonwoodhull** [comment]
Noticed this is still not working, looked back at the history. Look at these consecutive commits on init.js:

```
commit 1175ada94d5176ca8150ada118ae75ebd2deaab7
Author: shane <shane@freshclickmedia.com>
Date:   Fri May 13 11:35:45 2016 +0100

    fixes #1971 - undo and redo are not available in view mode

diff --git a/htdocs/js/ui/init.js b/htdocs/js/ui/init.js
index 98d9437..fe5303b 100644
--- a/htdocs/js/ui/init.js
+++ b/htdocs/js/ui/init.js
@@ -141,6 +141,7 @@ RCloud.UI.init = function() {
             ['command', 'z'],
             ['ctrl', 'z']
         ],
+        modes: ['writeable'],
         action: function() { editor.step_history_undo(); }
     }, {
         category: 'Notebook Management',
@@ -150,6 +151,7 @@ RCloud.UI.init = function() {
             ['ctrl', 'y'],
             ['command', 'shift', 'z']
         ],
+        modes: ['writeable'],        
         action: function() { editor.step_history_redo(); }
     }]);
 

commit 2c6fdbcc6947a15e9bebb0b2b72451a6ad38293e
Author: shane <shane@freshclickmedia.com>
Date:   Thu May 5 16:33:28 2016 +0100

    fixes #1899, history step forwards also enabled

diff --git a/htdocs/js/ui/init.js b/htdocs/js/ui/init.js
index 8276476..98d9437 100644
--- a/htdocs/js/ui/init.js
+++ b/htdocs/js/ui/init.js
@@ -141,7 +141,6 @@ RCloud.UI.init = function() {
             ['command', 'z'],
             ['ctrl', 'z']
         ],
-        modes: ['writeable'],
         action: function() { editor.step_history_undo(); }
     }, {
         category: 'Notebook Management',
@@ -151,7 +150,6 @@ RCloud.UI.init = function() {
             ['ctrl', 'y'],
             ['command', 'shift', 'z']
         ],
-        modes: ['writeable'],
         action: function() { editor.step_history_redo(); }
     }]);
```

This says to me that it's probably not the right condition and we may need to look at mode here as well as writeability.

**Comment by gordonwoodhull** [comment]
These contradictory commits are definitely a consequence of our nonlinear workflow of having tons of PRs open. I apologize for that.

**Comment by gordonwoodhull** [comment]
Fixed in #2059
